frontend/views/pages/Payments.vue

```diff
@@ -42,44 +42,51 @@ page(
           :class='{ "is-active": ephemeral.activeTab === link.url}'
           :data-test='`link-${link.url}`'
           :aria-expanded='ephemeral.activeTab === link.url'
-          @click='ephemeral.activeTab = link.url'
+          @click='handleTabClick(link.url)'
         )
           | {{ link.title }}
           span.tabs-notification(v-if='link.notification') {{ link.notification }}
 
-      // Remove condition -> the search should be available for every tab if more than x result
       search(
-        v-if='needsIncome || ephemeral.activeTab !== "PaymentRowTodo"'
+        v-if='paymentsListData.length && ephemeral.activeTab !== "PaymentRowTodo"'
         :placeholder='L("Search payments...")'
         :label='L("Search for a payment")'
-        v-model='ephemeral.searchText'
+        v-model='form.search'
       )
 
       .tab-section
-        .c-container(v-if='hasPayments')
+        .c-container(v-if='paymentsFiltered.length')
           payments-list(
             :titles='tableTitles'
-            :paymentsList='paymentsListData'
+            :paymentsList='paginateList(paymentsFiltered)'
             :paymentsType='ephemeral.activeTab'
           )
           .c-footer
-            .c-payment-record(v-if='!needsIncome && ephemeral.activeTab === "PaymentRowTodo"')
+            .c-payment-record(v-if='ephemeral.activeTab === "PaymentRowTodo"')
               i18n.c-payment-info(
                 tag='b'
                 data-test='paymentInfo'
-                :args='tableFooterStatus'
+                :args='footerTodoStatus'
               ) {amount} in total, to {count} members
 
               i18n.button(
                 tag='button'
                 data-test='recordPayment'
                 @click='openModal("RecordPayment")'
               ) Record payments
-            payments-pagination(v-else)
-
-        .c-container-empty(v-else)
+            payments-pagination(
+              v-else
+              :count='paymentsFiltered.length'
+              :rowsPerPage='ephemeral.rowsPerPage'
+              :page.sync='ephemeral.currentPage'
+              @changePage='handlePageChange'
+              @changeRowsPerPage='handleRowsPerPageChange'
+            )
+        .c-container-noresults(v-else-if='paymentsListData.length && !paymentsFiltered.length' data-test='noResults')
```

Comment:
Clever! I've always checked the presence of the search string. Your solution feels cleaner.

Reply:
Thank u!